python/src/iceberg/transforms.py

```diff
@@ -0,0 +1,567 @@
+        if self == other:
+            return True
+        elif isinstance(self._type, StringType) and isinstance(other, TruncateTransform) and isinstance(other._type, StringType):
+            return self._width >= other._width
+
+        return False
+
+    def to_human_string(self, value) -> str:
+        if value is None:
+            return "null"
+        elif isinstance(self._type, BinaryType):
+            return transform_util.base64encode(value)
+        else:
+            return str(value)
+
+
+class UnknownTransform(Transform):
+    """A transform that represents when an unknown transform is provided
+
+    Args:
+      source_type (Type): An Iceberg `Type`
+      transform (str): A string name of a transform
+
+    Raises:
+      AttributeError: If the apply method is called.
+    """
+
+    def __init__(self, source_type: IcebergType, transform: str):
+        super().__init__(
+            transform,
+            f"UnknownTransform(source_type={repr(source_type)}, transform={repr(transform)})",
+        )
+        self._type = source_type
+        self._transform = transform
+
+    def apply(self, value):
+        raise AttributeError(f"Cannot apply unsupported transform: {self}")
+
+    def can_transform(self, target: IcebergType) -> bool:
+        return self._type == target
+
+    def result_type(self, source: IcebergType) -> IcebergType:
+        return StringType()
+
+
+class VoidTransform(Transform):
+    """A transform that always returns None"""
+
+    _instance = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(VoidTransform, cls).__new__(cls)
+        return cls._instance
+
+    def __init__(self):
+        super().__init__("void", "transforms.always_null()")
+
+    def apply(self, value):
+        return None
+
+    def can_transform(self, target: IcebergType) -> bool:
+        return True
+
+    def to_human_string(self, value) -> str:
+        return "null"
+
+
+_HAS_WIDTH = re.compile("(\\w+)\\[(\\d+)\\]")
+
+
+def from_string(source_type: IcebergType, transform: str) -> Transform:
+    transform_lower = transform.lower()
+    match = _HAS_WIDTH.match(transform_lower)
+
+    if match is not None:
+        name = match.group(1)
+        w = int(match.group(2))
+        if name == "truncate":
+            return TruncateTransform(source_type, w)
+        elif name == "bucket":
+            return BaseBucketTransform(source_type, w)
+
+    if transform_lower == "identity":
+        return identity(source_type)
+
+    try:
+        return DateTimeTransform(source_type, transform_lower)
+    except (KeyError, ValueError) as e:
+        pass  # fall through to return unknown transform
+
+    if transform_lower == "void":
+        return VoidTransform()
+
+    return UnknownTransform(source_type, transform)
+
+
+def identity(source_type: IcebergType) -> IdentityTransform:
+    return IdentityTransform(source_type)
```

Comment:
For these one-liner functions, the one line doesn't seem all that complicated. Would it be worth it to get rid of the indirection and just call `IdentityTransform(...)` when needed instead of `identity(...)` or as another example, call `DateTimeTransform(source_type, "year")` directly instead of `year(...)`. I guess I'm not completely sold on the added value of the wrapper function.